server/src/main/kotlin/org/octopusden/octopus/dms/service/impl/ComponentServiceImpl.kt

```diff
@@ -51,6 +51,34 @@ class ComponentServiceImpl(
         }
     }
 
+    /**
+     * Update component name and return new component name
+     * @param componentName - old component name
+     * @param newComponentName - new component name
+     * @return new component name
+     * @throws NotFoundException if component with name [componentName] not found in releng
+     * @throws NoSuchElementException if component with name [componentName] not found in database
+     */
+    @Transactional(readOnly = false)
+    override fun updateComponentName(componentName: String, newComponentName: String): String {
+        var result = componentName
+        log.debug("Update component name from '$componentName' to '$newComponentName'")
+        componentRepository.lock(componentName.hashCode())
```

Comment:
Sorry, I'm not familiar with DMS, but I'll ask just in case - usually after the LOCK operation and the logic part there is an UNLOCK operation. Have you forgotten about this?

Reply:
Fixed

Reply:
А что нам даст лок на запись в таблице? в ней только  id и name.  от чего защищаемся?